app/src/main/java/com/siddharth/practiceapp/ui/fragments/HomeFragment.kt

```diff
@@ -96,15 +93,15 @@ class HomeFragment : Fragment(R.layout.fragment_home) {
             if (it is Response.Success) {
                 (requireActivity() as MainActivity).hideSideBar()
                 Log.d(TAG, "size of homeDataList from db is ${it.data!!.size}")
-                adapter.dataList.clear()
-                adapter.dataList.addAll(it.data)
+                //adapter.dataList.clear()
+                adapter.submitList(it.data)
                 adapter.notifyItemRangeChanged(0, it.data.size)
```

Comment:
sorry I am not familiar with this but is notifyItemRangeChanged required in listAdapter also?

Reply:
No its need. I will remove it.